### DISCUSSION TITLE
TimescaleDB extension:  Continuous aggregates are not supported

### DISCUSSION THREAD

**Discussion by samfweb** [supabase/supabase#19475] [Questions]
# Bug report

## Describe the bug

Attempting to create a continuous aggregate using the TimescaleDB extension (https://docs.timescale.com/getting-started/latest/create-cagg/create-cagg-basics/) on a cloud-hosted Supabase database fails due to lack of support under current Apache license. 

This is a core feature of TimescaleDB.

## To Reproduce

1. Follow the TimescaleDB 'Getting started' tutorial at https://docs.timescale.com/getting-started/latest/
2. Attempt to create a continuous aggregate with the script below
3. See error: `Failed to run sql query: functionality not supported under the current "apache" license`

Script from step 2:
```sql
CREATE MATERIALIZED VIEW stock_candlestick_minutes
WITH (timescaledb.continuous) AS
SELECT
  time_bucket('1 minute', "time") AS minute,
  symbol,
  max(price) AS high,
  min(price) AS low
FROM stocks_real_time srt
GROUP BY minute, symbol;
```

## Expected behavior

The continuous aggregate is created.

## Screenshots

![image](https://user-images.githubusercontent.com/44712417/218231273-152d3f79-6658-472c-860b-6cb29b39fb38.png)

## System information

- Current cloud defaults

**Comment by samfweb** [comment]
I realise this is a licensing issue: https://docs.timescale.com/timescaledb/latest/timescaledb-edition-comparison/ 

Are we able to run community edition if we host Supabase ourselves?

**Comment by DaleBeckles** [comment]
I just experienced a similar issue to @samfweb relating to the TimescaleDB extension when I attempted to add a data retention policy using the command:

`SELECT add_retention_policy('table_name', INTERVAL '100 days')`

The result/error returned was:

 `Failed to run sql query: function "add_retention_policy" is not supported under the current "apache" license`

Is there a way to run the Community Edition of Timescale in your hosted offering (I am a Pro user)?? We need to run multiple instances of Supabase with Timescale in different geographic regions re: data privacy.

Limitations of the TimescaleDB licensing issues need to be added to the doc.

**Comment by guiofranca** [comment]
As Timescale License says, you can't use the community edition as an offer of your product, just as an Added Value. So, if you are using Timescale through Supabase, then Supabase can`t offer you the full community edition, just the apache license capabilities.

I believe that to get full functionality of Timescale, you would have to start from Postgres, and then install Timescale and Supabase extensions on top of it. Obviously as self hosted solution. You could also try to get lower access to postgres shipped with supabase and install timescaledb from there, but never from the supabase studio interface.

**Reply by bitnom** [reply]
What is "An offer of your product?" Supabase isn't selling a "Timescaledb" package or anything like that. It looks like added value to me.

**Comment by bitnom** [comment]
Here is how to resolve it: https://www.timescale.com/forum/t/upgrade-your-license-to-timescale-to-use-this-free-community-feature/677/2

I need this fixed in my supabase cloud. This is a nightmare I didn't realize. There must be a way for me to enable the Community license in my Supabase cloud instance. Supabase isn't selling it in this case, right? It's an extension I'm requesting. Supabase is my host. Allow me to utilize my own license of the extension, please.

**Comment by servrox** [comment]
Wrote a thread about it + a proper solution! I used a github action to re-build the image.
https://x.com/servrox1337/status/2023679230530211958

**Comment by Rubenburdin** [comment]
theres a workaround that gets you pretty close to continuous aggregates using what's already available. You can create a regular materialized view with the same query (just drop the WITH (timescaledb.continuous) part) and then use Supabase's pg_cron extension to refresh it on a schedule. Something like SELECT cron.schedule('refresh_candlesticks', '*/1 * * * *', 'REFRESH MATERIALIZED VIEW CONCURRENTLY stock_candlestick_minutes') to refresh every minute.

The tradeoff is that a regular materialized view recomputes the entire result set on each refresh instead of only processing new data like a true continuous aggregate would. For smaller datasets or wider time buckets (hourly, daily) its honestly fine. If your table gets large you can add a unique index on (minute, symbol) so the CONCURRENTLY option works, which means reads arent blocked during the refresh.
